### PR TITLE
Update dashboard guide

### PR DESCRIPTION
**Pull request by balloob**
## Description:
Update the ESPHome dashboard add-on getting started guide

https://deploy-preview-1233--esphome.netlify.app/guides/getting_started_hassio.html

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** https://github.com/esphome/esphome/pull/1887

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: guides/getting_started_hassio.rst [hunk 1]
```diff
@@ -10,27 +10,17 @@ your ESP8266/ESP32 boards. In this guide we’ll go through how to setup a
 basic "node" by use of the Home Assistant add-on.
 
 But first, here's a very quick introduction to how ESPHome works:
-ESPHome is a *tool* which reads in a YAML configuration file (just like Home Assistant)
-and creates a custom firmware binary. The tool also has many helpers that simplify flashing devices (uploading the new binary file)
-and aim to make managing your ESP boards as simple as possible. Once you have added devices
-or sensors in ESPHome's configuration, they will even automatically show up in Home
-Assistant's UI.
+ESPHome is a *tool* which aims to make managing your ESP boards as simple as possible. It reads in a YAML configuration file (just like Home Assistant) and creates a custom firmware binary which it installs on your ESP device. Devices or sensors added in ESPHome's configuration will automatically show up in Home Assistant's UI.
```
Comment (frenck) [inline guides/getting_started_hassio.rst]:
```suggestion
ESPHome is a *tool* which aims to make managing your ESP boards as simple as possible. It reads in a YAML configuration file (just like Home Assistant) and creates custom firmware which it installs on your ESP device. Devices or sensors added in ESPHome's configuration will automatically show up in Home Assistant's UI.
```

File: guides/getting_started_hassio.rst [hunk 2]
```diff
@@ -44,6 +34,20 @@ creating your first configuration.
     :align: center
     :width: 95.0%
 
+The wizard will guide you through creating your first configuration and, depending on your browser, install it on your device. You will need to name your configuration and enter your wireless network settings so that the ESP device can come online and can communicate with Home Assistant.
+
+.. note::
+
+  The most difficult part of a new ESPHome device is the initial installation. Installation requires that your ESP device is connected with a cable to your computer. Once the initial installation is done, future updates can be applied wirelessly.
+
+  If you use `Microsoft Edge <https://www.microsoft.com/edge>`_ or `Google Chrome <https://www.google.com/chrome>`_, you will be able to install your new configuration via your browser to your ESP device, saving you the hassle of connecting the ESP devices to the machine running Home Assistant.
```
Comment (frenck) [inline guides/getting_started_hassio.rst]:
```suggestion
  If you use `Microsoft Edge <https://www.microsoft.com/edge>`_ or `Google Chrome <https://www.google.com/chrome>`_, you will be able to install the initial configuration via your browser onto your ESP device, saving you the hassle of connecting the ESP devices to the machine running Home Assistant.
```

File: guides/getting_started_hassio.rst [hunk 3]
```diff
@@ -106,8 +103,11 @@ anything really, for example lights) and is connected to the pin ``GPIO5``.
 Connecting your device to Home Assistant
 ----------------------------------------
 
-Now when you go to the Home Assistant "Integrations" screen (under "Configuration" panel), you
-should see the ESPHome device show up in the discovered section (although this can take up to 5 minutes).
+Once your configuration is installed on your ESP device and is online, it will be automatically discovered by Home Assistant and offered to set up on your integrations screen:
+
+.. image:: https://my.home-assistant.io/badges/integrations.svg
+   :target: https://my.home-assistant.io/redirect/integrations/
```
Comment (frenck) [inline guides/getting_started_hassio.rst]: Maybe start a config flow instead?

```suggestion
   :target: https://my.home-assistant.io/redirect/config_flow_start/?domain=esphome
```

File: guides/getting_started_hassio.rst [hunk 4]
```diff
@@ -106,8 +103,11 @@ anything really, for example lights) and is connected to the pin ``GPIO5``.
 Connecting your device to Home Assistant
 ----------------------------------------
 
-Now when you go to the Home Assistant "Integrations" screen (under "Configuration" panel), you
-should see the ESPHome device show up in the discovered section (although this can take up to 5 minutes).
+Once your configuration is installed on your ESP device and is online, it will be automatically discovered by Home Assistant and offered to set up on your integrations screen:
+
+.. image:: https://my.home-assistant.io/badges/integrations.svg
```
Comment (balloob) [inline guides/getting_started_hassio.rst]:
```suggestion
.. image:: https://my.home-assistant.io/badges/config_flow_start.svg
```

File: guides/getting_started_hassio.rst [hunk 5]
```diff
@@ -44,6 +34,24 @@ creating your first configuration.
...
 
+The wizard will guide you through creating your first configuration and, depending on your browser, install it on your device. You will need to name your configuration and enter your wireless network settings so that the ESP device can come online and can communicate with Home Assistant.
+
+.. raw:: html
+
+    <a name='webserial'></a>
+
+.. note::
+
+    The most difficult part of setting a new ESPHome device is the initial installation. Installation requires that your ESP device is connected with a cable to a computer. Later updates can be installed wirelessly.
```
Comment (frenck) [inline guides/getting_started_hassio.rst]:
```suggestion
    The most difficult part of setting up a new ESPHome device is the initial installation. Installation requires that your ESP device is connected with a cable to a computer. Later updates can be installed wirelessly.
```

**Comment by glmnet** [comment]
This doc uses the install term for both installing the add on and later install means flashing the device. Is this really desired?

**Comment by glmnet** [comment]
Also is the first time in the microcontroller world that install means flashing/downloading/update firmware. Coming up with a term like this for a well known activity is a bit odd to me.

**Comment by balloob** [comment]
> flashing/downloading/update firmware

ESPHome actually uses "upload".

Installing is a common term for getting some code on a device. It's used on desktop, phones. Users get it.

Also this Quora looks like it leans towards installing:

https://www.quora.com/What-is-the-difference-between-flashing-rooting-and-installing-firmware

> Flashing is writing data to nonvolatile flash memory.
> 
> Installing firmware is writing data/software to nonvolatile memory that is typically executed when the device powers up. Since the only flash memory on systems with a disk drive is probably firmware, that amounts to the same thing.

**Comment by balloob** [comment]
I've updated the add-on installation header. Thinking now, we also use dashboard and add-on interchangeably. Probably should address that too.

**Comment by glmnet** [comment]
> 
> ESPHome actually uses "upload".

Indeed dropping upload is great as that is too cloudy

> 
> Installing is a common term for getting some code on a device. It's used on desktop, phones. Users get it.
> 

Agree, I might just wait before giving more feedback. I find it a bit confusing as you don't install over and over stuff like you normally do with ESPHome. Generally you install something and then switch your mind to configure it. 

Anyway if upload is erased from everywhere and you can only install then you cannot get confused. 

Overall is looking great!

File: guides/getting_started_hassio.rst [hunk 6]
```diff
@@ -2,36 +2,25 @@ Getting Started with ESPHome and Home Assistant
...
-.. figure:: images/hassio_repo.png
+Installing ESPHome Dashboard
+----------------------------
 
-Click on ESPHome, then INSTALL.
+The ESPHome Dashboard can be installed as a Home Assistant add-on, which you can find in the add-on store in the Supervisor panel. Open it using the following button then click on INSTALL:
 
-.. figure:: images/hassio_addons_section.png
+.. image:: https://my.home-assistant.io/badges/supervisor_addon.svg
+   :target: https://my.home-assistant.io/redirect/supervisor_addon/?addon=a0d7b954_esphome
```
Comment (jesserockz) [inline guides/getting_started_hassio.rst]: Can we open this in a new tab? This currently will take the user away from the guide...

File: guides/getting_started_hassio.rst [hunk 7]
```diff
@@ -44,11 +33,29 @@ creating your first configuration.
...
+
 Dashboard Interface
 -------------------
 
 Assuming you created your first configuration file with the wizard, let's take a quick
-tour of the ESPHome dashboard interface.
+tour of the ESPHome Dashboard interface.
 
 .. figure:: images/dashboard.png
     :align: center
```
Comment (jesserockz) [inline guides/getting_started_hassio.rst:62]: Might need to update the screenshots, particularly this one that has `upload`

**Review by jesserockz — APPROVED** [review]

**Pull request merged**